packages/flame/lib/src/geometry/circle_component.dart

```diff
@@ -30,8 +32,19 @@ class CircleComponent extends ShapeComponent implements SizeProvider {
     super.angle,
     super.anchor,
     super.paint,
+    super.paintLayers,
   }) : super(size: Vector2.all(relation * min(parentSize.x, parentSize.y)));
 
+  @override
+  @mustCallSuper
+  Future<void> onLoad() async {
+    void updateCenterOffset() => _centerOffset = Offset(size.x / 2, size.y / 2);
+    size.addListener(updateCenterOffset);
```

Comment:
This should be a different PR so we could have change log entry specifically for it. Wdyt?

Reply:
It is related to this PR though, it's just making it more efficient to render the paint layers (and it's all internal, so a separate changelog entry isn't really needed).

Reply:
won't this make the circle component be in fact resizable? Because iirc, none of the shape components are right now.

Or maybe I am confusing something...

Reply:
It was already resizable before (all components having the `SizeProvider` mixin are proven to work with resizing), the only thing this does is storing the center point so that it can render from the center. But I see now that it doesn't take the anchor into consideration, so good that we started discussing here. :sweat_smile:

Reply:
Alright, so maybe I am confusing with some old bug. Gonna approve it!

Reply:
It seems like the render function was already wrong before. I'll merge this and fix the other bug tomorrow in a separate PR, for a better changelog. ;)